server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/EndpointSubsystemReader_9_0.java

```diff
@@ -597,4 +674,75 @@ private void parseSni(final XMLExtendedStreamReader reader, final ModelNode encr
       ParseUtils.requireNoContent(reader);
       operations.add(sniOp);
    }
+
+   private void parsePrefix(final XMLExtendedStreamReader reader, final ModelNode prefix, final List<ModelNode> operations) throws XMLStreamException {
```

Comment:
Yes I totally agree... I tried to refactor the parser code at first but failed miserably...

Answering the question - it's called parse `prefix` because it parses `<prefix>` xml element. I noticed that the parser (semi?)follows this convention. 

I also created [ISPN-6858](https://issues.jboss.org/browse/ISPN-6858) for @ryanemerson and I would kindly ask for preparing a plan for refactoring the parser part (I would be happy to implement a part of that but I need a lot of guidance).